### ISSUE TITLE
Postgres: Add support for ALTER VIEW statement

### ISSUE THREAD

**Issue by derickl** [sqlfluff/sqlfluff#2094]
## Expected Behaviour
SQLFluff should be able to parse `ALTER VIEW` statements

## Observed Behaviour
```shell
sqlfluff lint --dialect postgres -n sample.sql

== [sample.sql] FAIL                                                                                                                                                                                                                  
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'ALTER
                       | VIEW myview ALTER date_column SET ...'
```

## Steps to Reproduce
1. Create `sample.sql` using the following query
```sql
ALTER VIEW myview ALTER date_column SET DEFAULT NOW();
```
2. sqlfluff lint --dialect postgres -n sample.sql
3. sqlfluff parse --dialect postgres -n sample.sql

## Dialect
`postgres`

## Version
`sqlfluff, version 0.8.2 and master`
`Python 3.8.22`

## Configuration
```
N/A : Default configuration
```
